egs2/slue-voxpopuli/slu1/local/prepare_nel_data.py

```diff
@@ -0,0 +1,169 @@
+                        else:
+                            if label != "":
+                                entity_phrase += " " + label
+                            entity_end_time = end_time
+                        all_word_alignments[utt_id][idx] = (
+                            "#" + label,
+                            start_time,
+                            end_time,
+                        )
+                    entity_alignments[utt_id].append(
+                        [entity_phrase, entity_start_time, entity_end_time]
+                    )
+                    curr_idx += len_phrase
+                    data_idx += 1
+                else:
+                    curr_idx += 1
+                if data_idx < len(gt_labels):
+                    wrd_lst, len_phrase = read_gt_sample(gt_labels, data_idx)
+            else:
+                curr_idx += 1
+            if curr_idx == len(all_word_alignments[utt_id]) and data_idx != len(
+                gt_labels
+            ):
+                import pdb
+
+                pdb.set_trace()
+                print(data_idx, len(gt_labels))
+                print(gt_labels)
+                print(text)
+                sys.exit("Process exited, possibly an issue with text processing.")
+
+
+def modify_word_alignments(dataset_obj, data_dir, data_split, extract_gt):
+    """
+    Modify the word alignments to mark entity phrases for evaluation
+    """
+    all_word_alignments = {}
+    entity_alignments = {}
+    for idx, _ in enumerate(tqdm(dataset_obj)):
+        gt_labels = []
+        utt_id = dataset_obj[idx]["id"]
+        all_word_alignments[utt_id] = []
+        wrd_durs = dataset_obj[idx]["word_timestamps"]
+        num_wrds = len(wrd_durs["word"])
+        nel_labels = dataset_obj[idx]["ne_timestamps"]
+        num_ne = len(nel_labels["ne_label"])
+        if num_ne != 0:
+            txt = dataset_obj[idx]["text"]
+            for ne_idx in range(num_ne):
+                start_id = nel_labels["start_char_idx"][ne_idx]
+                length = nel_labels["char_offset"][ne_idx]
+                phrase = txt[start_id : start_id + length]
+                gt_labels.append((phrase, start_id, length))
+        for wrd_idx in range(num_wrds):
+            word = wrd_durs["word"][wrd_idx]
+            start_sec = wrd_durs["start_sec"][wrd_idx]
+            end_sec = wrd_durs["end_sec"][wrd_idx]
+            all_word_alignments[utt_id].append([word, start_sec, end_sec])
+        update_alignment_dct(all_word_alignments, entity_alignments, utt_id, gt_labels)
+
+    if data_split == "validation":
+        data_split = "devel"  # to match ESPNET format
+    save_json(
+        os.path.join(data_dir, f"{data_split}_all_word_alignments.json"),
+        all_word_alignments,
+    )
+    if extract_gt:
+        save_json(
+            os.path.join(data_dir, f"{data_split}_entity_alignments.json"),
+            entity_alignments,
+        )
+
+
+def main(data_dir="data/nel_gt", is_blind=False):
```

Comment:
optional: Remove all conditions concerning blind NEL test set. 
1. remove `is_blind` argument from here 
2. remove line 164
3. remove `extract_gt` argument from `modify_word_alignments` lines 165, 118, and 152.

Reply:
Fixed